ImageData/Hdf5Loader.h

```diff
@@ -362,21 +370,21 @@ std::map<CARTA::StatsType, std::vector<double>>* Hdf5Loader::GetRegionSpectralDa
             sum_sq[z] = 0;
         }
 
+        RegionState region_state = _region_states[region_id];
+
         // Load each X slice of the swizzled region bounding box and update Z stats incrementally
         for (size_t x = 0; x < num_x; x++) {
-            casacore::Slicer slicer;
-            if (_num_dims == 4) {
-                slicer = casacore::Slicer(IPos(4, 0, y_min, x + x_min, stokes), IPos(4, num_z, num_y, 1, 1));
-            } else if (_num_dims == 3) {
-                slicer = casacore::Slicer(IPos(3, 0, y_min, x + x_min), IPos(3, num_z, num_y, 1));
+            if (!_connected) {
```

Comment:
See comment next to these properties. This could be a function on the frame object which uses the frame's original state variables.